src/search/hybridRetriever.ts

```diff
@@ -122,11 +108,12 @@ export class HybridRetriever extends BaseRetriever {
   private async rewriteQuery(query: string): Promise<string> {
     try {
       const promptResult = await this.queryRewritePrompt.format({ question: query });
-      const rewrittenQueryObject = await this.llm.invoke(promptResult);
+      const chatModel = ChatModelManager.getInstance().getChatModel();
+      const rewrittenQueryObject = await chatModel.invoke(promptResult);
 
       // Directly return the content assuming it's structured as expected
       if (rewrittenQueryObject && "content" in rewrittenQueryObject) {
-        return rewrittenQueryObject.content;
+        return rewrittenQueryObject.content as string;
```

Comment:
there is a type issue here but this refactor doesn't have any logic change. It should be safe to cast to string and keep the same type definition as before.